src/vs/workbench/parts/extensions/electron-browser/extensionTipsService.ts

```diff
@@ -110,46 +110,43 @@ export class ExtensionTipsService extends Disposable implements IExtensionTipsSe
 	private processWorkspaceRecommendations(extensionsContent: IExtensionsContent): TPromise<string[]> {
 		const regEx = new RegExp(EXTENSION_IDENTIFIER_PATTERN);
 
-		if (extensionsContent && extensionsContent.recommendations) {
+		if (extensionsContent && extensionsContent.recommendations && extensionsContent.recommendations.length) {
+			let countBadRecommendations = 0;
+			let badRecommendationsString = '';
 			let filteredRecommendations = extensionsContent.recommendations.filter((element, position) => {
-				return extensionsContent.recommendations.indexOf(element) === position && regEx.test(element);
-			});
-
-			if (filteredRecommendations.length !== extensionsContent.recommendations.length) {
-				let badlyFormattedRecommendations = extensionsContent.recommendations.filter(element => {
-					return filteredRecommendations.indexOf(element) === -1;
-				});
-
-				let badRecommendationsString = '';
-				badlyFormattedRecommendations.forEach(element => {
-					badRecommendationsString += element + '\n';
-				});
+				if (!regEx.test(element)) {
```

Comment:
This way if there is a recommendation not following the regex appearing twice, then it will appear in the `badRecommendationsString` twice.

Duplication in recommendations doesn't cause any harm. We don't need to add a warning for it.

Reply:
I'm fixing this to only log once per distinct value as recommended. However, the reason I was in favor of logging duplicates is specifically for when the author has duplicate bad entries. They will see they have a bad entry, fix it, then see the exact same error logging and possibly be very confused if their changes were picked up. I could check to see if a bad entry is a duplicate (using lastIndexOf, and log as (bad format, duplicate)